File: app/src/main/java/com/eugene_poroshin/money_manager/operations/OperationTypeConverter.java
package com.eugene_poroshin.money_manager.operations;

import androidx.room.TypeConverter;

public class OperationTypeConverter {

    @TypeConverter
    public static String fromTypeToString(OperationType type) {
        if (type == null)
            return null;
        return type.toString();
    }

    @TypeConverter
    public static OperationType fromStringToType(String type) {
        if (type.equals(OperationType.CONSUMPTION.toString())) {
            return OperationType.CONSUMPTION;
        } else if (type.equals(OperationType.INCOME.toString())) {
            return OperationType.INCOME;
        } else {
            throw new IllegalArgumentException("Could not recognize type");
        }
    }
}

File: app/src/main/java/com/eugene_poroshin/money_manager/operations/OperationTypeConverter.kt
package com.eugene_poroshin.money_manager.operations

import androidx.room.TypeConverter

object OperationTypeConverter {
    @JvmStatic
    @TypeConverter
    fun fromTypeToString(type: OperationType?): String? {
        return type?.toString()
    }

    @JvmStatic
    @TypeConverter
    fun fromStringToType(type: String): OperationType {
        return when (type) {
            OperationType.CONSUMPTION.toString() -> OperationType.CONSUMPTION
            OperationType.INCOME.toString() -> OperationType.INCOME
            else -> throw IllegalArgumentException("Could not recognize type")
        }
    }
}